cmd/translate/main.go

```diff
@@ -1,40 +1,91 @@
 package main
 
 import (
+	"context"
 	"fmt"
+	"log"
+	"net"
+	"net/http"
 	"os"
+	"time"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	pb "go.expect.digital/translate/pkg/server/translate/v1"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/soheilhy/cmux"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
 )
 
+type TranslateServiceServer struct {
+	pb.UnimplementedTranslateServiceServer
+}
+
 var cfgFile string
 
 // rootCmd represents the base command when called without any subcommands.
 var rootCmd = &cobra.Command{
 	Use:   "translate",
 	Short: "Enables translation for Cloud-native systems",
 	Long:  `Enables translation for Cloud-native systems`,
-	// Uncomment the following line if your bare application
-	// has an action associated with it:
-	// Run: func(cmd *cobra.Command, args []string) { },
-}
+	Run: func(cmd *cobra.Command, args []string) {
+		grpcSever := grpc.NewServer()
+		mux := runtime.NewServeMux()
+		err := pb.RegisterTranslateServiceHandlerFromEndpoint(
+			context.Background(),
+			mux,
+			"localhost:8080",
+			[]grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())})
+		if err != nil {
+			log.Panic(err)
+		}
 
-func main() {
-	Execute()
+		server := http.Server{
+			Handler:           mux,
+			ReadHeaderTimeout: time.Minute,
+		}
+
+		l, err := net.Listen("tcp", "localhost:8080")
+		if err != nil {
+			log.Panic(err)
+		}
+
+		m := cmux.New(l)
+		// a different listener for HTTP1
+		httpL := m.Match(cmux.HTTP1Fast())
+		// a different listener for HTTP2 since gRPC uses HTTP2
+		grpcL := m.Match(cmux.HTTP2())
+
+		go func() {
+			if err := server.Serve(httpL); err != nil {
+				log.Fatal(err)
+			}
+		}()
+
+		go func() {
+			if err := grpcSever.Serve(grpcL); err != nil {
+				log.Fatal(err)
+			}
+		}()
+
+		if err := m.Serve(); err != nil {
+			log.Panic(err)
+		}
+	},
 }
 
 // Execute adds all child commands to the root command and sets flags appropriately.
-// This is called by main.main(). It only needs to happen once to the rootCmd.
-func Execute() {
+func main() {
 	if err := rootCmd.Execute(); err != nil {
 		os.Exit(1)
 	}
 }
 
 func init() {
 	cobra.OnInitialize(initConfig)
-	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is examples/translate.yml)")
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is WORKDIR/translate.yaml)")
```

Comment:
WORKDIR is very abstract and doesn't convey enough details. There are some better alternatives like `./translate` or `$PWD/translate` that people are familiar with.

```suggestion
	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is ./translate.yaml)")
```